repair/repair_task.hh

```diff
@@ -17,6 +17,17 @@ public:
         : tasks::task_manager::task::impl(module, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity), parent_id) {
         _status.progress_units = "ranges";
     }
+
+    virtual future<tasks::task_manager::task::progress> get_progress() const override {
+        unsigned i = 0;
+        tasks::task_manager::task::progress progress{};
+        while (i < _children.size()) {
+            progress.completed += co_await gather_child_completed(i);
+            ++i;
+        }
```

Comment:
this looks like a for loop to me.
How about:
```c__
    tasks::task_manager::task::progress progress{};
    for (unsigned i = 0; i < _children.size(); ++i) {
        progress.completed += co_await gather_child_completed(i);
    }
```

or even better, have `gather_child_completed` accept a `const foreign_task_ptr&`, and then the loop will be even simpler:
```c++
    for (const auto& child : _children) {
        progress.completed += co_await gather_child_completed(child);
    }
```

Reply:
Child tasks may still be pushed to `_children` in the meantime. Thus, foreach shouldn't be used. 

I will change the while loop to for and leave a comment about the above.

Reply:
still, sorry, but I don't like gather_child_completed given an index rather than a ref to child.
That said, if it has no other use, let's just open code it here.  I don't think it warrants a method (unless again if it would be reused elsewhere)